src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4186,7 +4186,9 @@ checkNullable:
                 ' TODO - GreenSepList knows its error count. Expose it instead of recomputing it.
                 If propertyParameters.Count = 0 Then
                     Dim unexpected = ResyncAt({SyntaxKind.AsKeyword, SyntaxKind.ImplementsKeyword, SyntaxKind.EqualsToken})
-                    closeParen = closeParen.AddTrailingSyntax(unexpected)
+                    If unexpected.Count > 0 Then
+                        closeParen = closeParen.AddTrailingSyntax(unexpected, ERRID.ERR_ExpectedEOS)
```

Comment:
>closeParen = closeParen.AddTrailingSyntax(unexpected, ERRID.ERR_ExpectedEOS) [](start = 24, length = 76)

I am not sure if the ResyncAt above is necessary and, as a consequence, whether this special error handling is necessary as well. First, I do not see us doing similar things in other places where ```ParseParameters``` is used. Second, at the very least the condition for the resync is questionable, zero parameters doesn't indicate any error situation on its own, therefore, we shouldn't be resyncing just because of that (resync is an attempt to recover from an error). Third, it looks like this declaration with parameters ```Public ReadOnly Property NumberOfResult3(x as Integer)  String Integer JohnDoe WwwIIWww Wow``` is parsed as expected and an appropriate error is reported. So, it feels like we are simply duplicating the work that is done elsewhere.

Reply:
@AlekseyTs That's very reasonable. But I can't continue debugging to get to the better solution. I can't open the compilers solution on my current machine due to very low specs. (I was working on an Azure VM, for which my subscription has ended :cry:)

Reply:
I removed the resync in a draft PR and CI passed. So applying the change here and will see whether the new test will pass or not..